compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2537,4 +2564,166 @@ public int hashCode() {
     private interface TypeEqualityPredicate {
         boolean test(BType source, BType target, Set<TypePair> unresolvedTypes);
     }
+
+    public boolean hasFillerValue(BType type) {
+        if (type.tag < TypeTags.RECORD) {
+            return true;
+        }
+        switch (type.tag) {
+            case TypeTags.MAP:
+            case TypeTags.ANY:
+                return true;
+            case TypeTags.ARRAY:
+                return checkFillerValue((BArrayType) type);
+            case TypeTags.FINITE:
+                return checkFillerValue((BFiniteType) type);
+            case TypeTags.UNION:
+                return checkFillerValue((BUnionType) type);
+            case TypeTags.OBJECT:
+                return checkFillerValue((BObjectType) type);
+            case TypeTags.RECORD:
+                return checkFillerValue((BRecordType) type);
+            case TypeTags.TUPLE:
+                BTupleType tupleType = (BTupleType) type;
+                return tupleType.getTupleTypes().stream().allMatch(eleType -> hasFillerValue(eleType));
+            default:
+                return false;
+        }
+    }
+
+    private boolean checkFillerValue(BObjectType type) {
+        BAttachedFunction initFunction = ((BObjectTypeSymbol) type.tsymbol).initializerFunc;
+        if (initFunction == null) {
+            if ((type.tsymbol.flags & Flags.ABSTRACT) == Flags.ABSTRACT) {
+                return false;
+            }
+        } else {
+            if (initFunction.symbol.getReturnType().getKind() == TypeKind.ERROR) {
```

Comment:
IINM, this can never be just error. It's either nil or a union with error. Just checking `initFunction.symbol.getReturnType().getKind() != TypeKind.NIL` should be enough?